src/devices/secplus_v2.c

```diff
@@ -0,0 +1,444 @@
+            break;
+        }
+
+    }
+
+    // Do was have what we need ??
+    if (fixed_1.bits_per_row[0] == 0 || fixed_2.bits_per_row[0] == 0) {
+        //  No?  Awww F'ck it then
+        if (decoder->verbose)
+            fprintf(stderr, "%s: DECODE_FAIL_SANITY\n", __func__);
+        return DECODE_FAIL_SANITY;
+    }
+
+    // Assemble rolling_1[] and rolling_2[] into rolling_digits[]
+    uint8_t rolling_digits[24] = {0};
+    uint8_t *r;
+
+    r    = rolling_digits;
+    *r++ = rolling_2[8];
+    *r++ = rolling_1[8];
+    for (int i = 4; i < 8; i++) {
+        *r++ = rolling_2[i];
+    }
+    for (int i = 4; i < 8; i++) {
+        *r++ = rolling_1[i];
+    }
+
+    for (int i = 0; i < 4; i++) {
+        *r++ = rolling_2[i];
+    }
+    for (int i = 0; i < 4; i++) {
+        *r++ = rolling_1[i];
+    }
+
+    // compute rolling_total from rolling_digits[]
+    uint32_t rolling_total = 0;
+    uint32_t rolling_temp  = 0;
+
+    for (int i = 0; i < 18; i++) {
+        rolling_temp = (rolling_temp * 3) + rolling_digits[i];
+        // fprintf(stderr, ">> %12d\t%d\n", rolling_temp, rolling_digits[i]);
+    }
+
+    // Max value = 2^28 (268435456)
+    if ( rolling_temp >= 0x10000000 ) {
+        return DECODE_FAIL_SANITY;
+    }
+
+    // value is 28 bits thus need to shift over 4 bit
+    rolling_total = reverse32(rolling_temp);
+    rolling_total = rolling_total >> 4;
+
+
+    // Assemble "fixed" data part
+    uint64_t fixed_total = 0;
+    uint8_t *bb;
+    bb = fixed_1.bb[0];
+    fixed_total ^= ((uint64_t)bb[0]) << 32;
+    fixed_total ^= ((uint64_t)bb[1]) << 24;
+    fixed_total ^= ((uint64_t)bb[2]) << 16;
+
+    bb = fixed_2.bb[0];
+    fixed_total ^= ((uint64_t)bb[0]) << 12;
+    fixed_total ^= ((uint64_t)bb[1]) << 4;
+    fixed_total ^= (bb[2] >> 4) & 0x0f;
+
+    // fprintf(stderr, "fixed_total = %lu\n", fixed_total);
+
+    // int button    = fixed_total >> 32;
+    // int remote_id = fixed_total & 0xffffffff;
+    char fixed_str[16];
+    char rolling_str[16];
+
+    // rolling_total is a 28 bit unsigned number
+    // fixed_totals is 40 bit in a uint64_t
+    snprintf(fixed_str, sizeof(fixed_str), "%lu", fixed_total);
+    snprintf(rolling_str, sizeof(rolling_str), "%u", rolling_total);
+
+    /* clang-format off */
+    data_t *data;
+    data = data_make(
+            "model",       "Model",    DATA_STRING, "Secplus-v2",
+            "button_id",   "Button-ID",    DATA_INT,    (fixed_total >> 32),
+            "remote_id",   "Remote-ID",    DATA_INT,    (fixed_total & 0xffffffff),
+            // "fixed",       "",    DATA_INT,    fixed_total,
+            "fixed",       "Fixed_Code",    DATA_STRING,    fixed_str,
+            "rolling",     "Rolling_Code",    DATA_STRING,    rolling_str,
+            NULL);
+    /* clang-format on */
+
+    decoder_output_data(decoder, data);
+    return 1;
+}
+
+static char *output_fields[] = {
+
+        // Common fields
```

Comment:
nit: I'd like to see this sorted in the same order as the `data_make`. That helps visually checking if everything is correct. (it is, so it's fine)